pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1284,11 +1284,20 @@ private void buildProducerAndAddTopic(Topic topic, long producerId, String produ
                              boolean userProvidedProducerName, TopicName topicName,
                              ProducerAccessMode producerAccessMode,
                              Optional<Long> topicEpoch, CompletableFuture<Producer> producerFuture){
+        // Skip acquiring the topic's lock within addProducer if the producer will fail creation anyway.
+        if (!isActive()) {
+            log.info("[{}] Connection is no longer active. Skipping creation for produceId={} to topic {}.",
+                    remoteAddress, topicName, producerId);
+            return;
```

Comment:
It seems that we need call `commandSender.sendErrorResponse` to send the response of this Producer command?
Also need complete `producerFuture` with exception.

Reply:
I agree that the paradigm is to complete the `producerFuture` exceptionally and then remove it from the `producers` map. We cannot send an error response here because the connection is closed.

Reply:
Right, that make sense.